### PR TITLE
fix URLs in JSON-LD tags

### PR DESCRIPTION
**Pull request by rav-pradhan**
### What

Updated the logic for building URLs for filterable and legacy dataset pages in the JSON-LD tags. Prior to the fix, the URL for filterable dataset pages would point to the dataset controller rather than the URL on the site

### How to review

- Check that code changes make sense
- Run the dataset controller `develop` branch and the renderer with `ENABLE_JSONLD_CONTROL` set to `true`
- Legacy and filterable dataset pages should now have URLs that match the patterns as specified in the updated unit tests

### Who can review

Anyone but me

### CODE REVIEW

File: assets/templates/partials/json-ld/base.tmpl [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <script type="application/ld+json">
     {
         "@context": "http://schema.org",
-        "name": {{ .Metadata.Title -}},
+        "name": {{ .Metadata.Title }},
         "publisher": {
         "@type": "Organization",
```
Comment (janderson2) [inline assets/templates/partials/json-ld/base.tmpl:7]: @rav-pradhan Small scope creep, but could we please change this to:

```suggestion
        "@type": "GovernmentOrganization",
```

File: assets/templates/partials/json-ld/dataset/legacy.tmpl [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
+{{ $URLPath := .URI }}
     {{ $length := len $data.Datasets }}{{ if gt $length 0 }}
         {{ $latestDataset := index $data.Datasets 0}}
         {{ $numberOfDownloads := len $latestDataset.Downloads }}
         "distribution": [
         {{ range $i, $el := $latestDataset.Downloads }} {
             "@type": "DataDownload",
             "encodingFormat": {{ .Extension }},
-            "contentUrl": {{ .URI }}
+            "contentUrl": {{concatenateStrings "www.ons.gov.uk/file?uri=" $URLPath "/current/" .URI }}
```
Comment (janderson2) [inline assets/templates/partials/json-ld/dataset/legacy.tmpl:10]: I think it would be better to use the `SiteDomain` here as well rather than being hard coded.

File: assets/templates/partials/json-ld/dataset/legacy.tmpl [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
             }{{ if notLastItem $numberOfDownloads $i }},{{end}}
         {{ end }}
     ],
     {{ end }}
-    "url": {{ buildURL .URI .SiteDomain }},
+    "url": {{ concatenateStrings "www." .SiteDomain $URLPath }},
```
Comment (janderson2) [inline assets/templates/partials/json-ld/dataset/legacy.tmpl:15]: When on welsh pages this should probably be using the `cy.` subdomain rather than the `www.` subdomain.

File: assets/templates/partials/json-ld/base.tmpl (hunk 1 above)
Comment (rav-pradhan) [reply assets/templates/partials/json-ld/base.tmpl]: Updated

File: assets/templates/partials/json-ld/dataset/legacy.tmpl (hunk 2 above)
Comment (rav-pradhan) [reply assets/templates/partials/json-ld/dataset/legacy.tmpl]: Have updated to use `SiteDomain` and also a `SubDomain` value derived from the site language

File: assets/templates/partials/json-ld/dataset/legacy.tmpl (hunk 3 above)
Comment (rav-pradhan) [reply assets/templates/partials/json-ld/dataset/legacy.tmpl]: Updated

File: assets/templates/partials/json-ld/dataset/filterable.tmpl [hunk 4]
```diff
@@ -1,15 +1,17 @@
...
+{{ $URLPath := .URI }}
 {{ $length := len $data.Version.Downloads }}
+{{ $SiteDomain := .SiteDomain }}
 {{ if gt $length 0 }}
         "distribution": [
         {{ range $i, $el := $data.Version.Downloads }} {
             "@type": "DataDownload",
             "encodingFormat": {{ .Extension }},
-            "contentUrl": {{ .URI }}
+            "contentUrl": {{concatenateStrings $SiteDomain .URI }}
```
Comment (janderson2) [inline assets/templates/partials/json-ld/dataset/filterable.tmpl:10]: I think we should probably add the `Subdomain` logic to this even though we don't support welsh here yet.  It will save us coming back when we do.

File: assets/templates/partials/json-ld/dataset/filterable.tmpl [hunk 5]
```diff
@@ -1,15 +1,17 @@
...
             }{{if notLastItem $length $i }},{{end}}
         {{ end }}
         ],
 {{ end }}
-"url": {{ .URI }},
+"url": {{ concatenateStrings .SiteDomain "/datasets/" .DatasetId }},
```
Comment (janderson2) [inline assets/templates/partials/json-ld/dataset/filterable.tmpl:15]: And same again here.

File: assets/templates/partials/json-ld/dataset/filterable.tmpl (hunk 4 above)
Comment (rav-pradhan) [reply assets/templates/partials/json-ld/dataset/filterable.tmpl]: Fair point; have updated

File: assets/templates/partials/json-ld/dataset/filterable.tmpl (hunk 5 above)
Comment (rav-pradhan) [reply assets/templates/partials/json-ld/dataset/filterable.tmpl]: Updated
